dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -73,57 +75,57 @@ private static DependencyNode resolveCompileTimeDependencies(Artifact rootDepend
     return resolveCompileTimeDependencies(rootDependencyArtifact, false);
   }
 
-  private static DependencyNode resolveCompileTimeDependencies(Artifact rootDependencyArtifact,
-      boolean includeProvidedScope)
+  private static DependencyNode resolveCompileTimeDependencies(
+      Artifact rootDependencyArtifact, boolean includeProvidedScope)
+      throws DependencyCollectionException, DependencyResolutionException {
+    return resolveCompileTimeDependencies(
+        ImmutableList.of(rootDependencyArtifact), includeProvidedScope);
+  }
+
+  private static DependencyNode resolveCompileTimeDependencies(
+      List<Artifact> dependencyArtifacts, boolean includeProvidedScope)
       throws DependencyCollectionException, DependencyResolutionException {
-    
-    String key = Artifacts.toCoordinates(rootDependencyArtifact);
 
+    boolean useCache = dependencyArtifacts.size() == 1;
```

Comment:
Caching is only needed for when one artifact is queried.

Reply:
Why? Please add a comment to the code explaining this.

Reply:
I'm not sure about this. If it's just an optimization please take it out. It doesn't really justify the added complexity. I put the caching in because it did have a measurable affect on performance; that is,  tests went from ~10 minutes to under a minute. But in general we don't want to make code more complex for optimization unless we know it's too slow and can show the fix improves it.